src/registry/api.py

```diff
@@ -480,6 +481,21 @@ def parse_data(predict: Prediction, data: List[dict]):
         )
 
     APILog.from_request(request)
+    prediction.save()
+
+    try:
+        scores = calculate_score(prediction.id)
+    except Exception:
+        prediction.delete()
+        return 422, {"message": "Failed to calculate Scores"}
+
+    prediction.mae = scores["mae"]
+    prediction.mse = scores["mse"]
+    prediction.crps = scores["crps"]
+    prediction.log_score = scores["log_score"]
+    prediction.interval_score = scores["interval_score"]
+    prediction.wis = scores["wis"]
```

Comment:
calculate score on prediction save, this replaces the post_save signal by a instant score calculation & error if fails to calculate. This fixes an reported error of missing scores right after a Prediction upload